llvm/lib/CodeGen/SelectionDAG/SelectionDAGISel.cpp

```diff
@@ -1441,11 +1441,21 @@ static void processSingleLocVars(FunctionLoweringInfo &FuncInfo,
   }
 }
 
+static bool shouldEnableFastISel(const Function &Fn) {
```

Comment:
I thought we had some kind of hook for this somewhere already; I would think this is a target specific decision

Reply:
OptLevelChanger ?

Reply:
Thanks for the pointers, I will have a look! It would be nice if this could be done on a per function level though, as we wouldn't want to disable fast isel for the entire module

Reply:
AFAICT all the TargetMachine hooks operate independently of the Function being lowered.

Regarding `OptLevelChanger`, it seems to be used to force the opt level to "None" when we have `skipPass(Fn) == true`, and  then it queries the TM to check if FastISel is enabled for O0.


That said, some of the debug messages inside `OptLevelChanger` makes me think we could add move the check from this patch in there. What do you think?

Reply:
It's worth a try, it's supposed to allow opt level override at the function level, which is precisely what we're after.

Reply:
@RKSimon @arsenm could you check if this new approach is what you had in mind?